src/test/java/sg/dex/starfish/integration/developerTC/InvokeServiceFree_20.java

```diff
@@ -1,410 +1,425 @@
+     * it support both SYNC and ASYNC
+     * this test case is for testing Sync behaviour SYNC
+     */
+    @Test
+    public void testPrimeSync_1() {
+
+        // input to the operation
+        Map<String, Object> metaMap = new HashMap<>();
+        metaMap.put("first-n", "11");
+
+        RemoteAgent agentI =RemoteAgent.create(ocean,did,remoteAccount);
+
+        // get asset form asset id of remote operation asset
+        Operation remoteOperation = (Operation) agentI.getAsset("0e48ad0c07f6fe87762e24cba3e013a029b7cd734310bface8b3218280366791");
+
+//        // response will have asset id as value which has the result of the operation
+        Map<String, Object> response = remoteOperation.invokeResult(metaMap);
+//
+        Map<String, Object> result = (Map<String, Object>) remoteOperation.getOperationSpec().get("results");
+
+
+
+        for (Map.Entry<String, Object> me : result.entrySet()) {
+            Map<String, Object> spec = (Map<String, Object>) me.getValue();
+            String type = (String) spec.get("type");
+            if (type.equals("asset")) {
+                Asset a = (Asset) response.get(me.getKey());
+                assertNotNull(a);
+            } else if (type.equals("json")) {
+                Object a = response.get(me.getKey());
+                assertNotNull(a);
+
+            }
+        }
+
+    }
+
+    /**
+     * TEST PRIME ::For this operation the input must be type JSON and the response will  be type asset.
+     * it support both SYNC and ASYNC
+     * this test case is for testing ASYNC behaviour SYNC
+     */
+    @Test
+    public void testPrimeAsync_1() {
+
+        // input to the operation
+        Map<String, Object> metaMap = new HashMap<>();
+        metaMap.put("first-n", "20");
+
+        // RemoteAgent agentS =RemoteAgent.create(ocean,didSurfer,remoteAccount);
+        RemoteAgent agentI =RemoteAgent.create(ocean,did,remoteAccount);
+
+        // get asset form asset id
+        Operation remoteOperation = (Operation) agentI.getAsset("0e48ad0c07f6fe87762e24cba3e013a029b7cd734310bface8b3218280366791");
+        // invoking the prime operation and will get the job associated
+        Job job = remoteOperation.invokeAsync(metaMap);
+
+        // waiting for job to get completed
+        Map<String, Object> remoteAsset = job.getResult();
+
+        // Map<String, Object> response = Params.formatResponse(remoteOperation, JSON.toMap(remoteAsset.toString()), agentI);
+
+
+        Map<String, Object> result = (Map<String, Object>) remoteOperation.getOperationSpec().get("results");
+
+        for (Map.Entry<String, Object> me : result.entrySet()) {
+            Map<String, Object> spec = (Map<String, Object>) me.getValue();
+            String type = (String) spec.get("type");
+            if (type.equals("asset")) {
+                RemoteDataAsset a = (RemoteDataAsset) remoteAsset.get(me.getKey());
```

Comment:
In general, might not be a remote asset? Use the DataAsset interface instead I think